app/ts/components/TransactionPage/SaveTransfer.tsx

```diff
@@ -0,0 +1,88 @@
+import { Signal, useSignal } from '@preact/signals'
+import { useEffect } from 'preact/hooks'
+import { FavoriteModel, isFavorite, useFavorities } from '../../store/favorites.js'
+
+type Props = {
+	show?: boolean
+	addFavoriteStore: Signal<Partial<FavoriteModel> | undefined>
+}
+
+export const SaveTransfer = ({ show, addFavoriteStore }: Props) => {
+	const isSaved = useSignal(false)
+	const { add } = useFavorities()
+
+	const saveTransfer = () => {
+		if (addFavoriteStore.value === undefined || !isFavorite(addFavoriteStore.value)) return
+
+		add(addFavoriteStore.value)
+		isSaved.value = true
+	}
+
+	useEffect(() => {
+		if (!show) return
+		addFavoriteStore.value = { ...addFavoriteStore.peek(), label: '' }
+	}, [show])
+
+	if (!isFavorite(addFavoriteStore.value)) return <></>
+
+	if (isSaved.value === true) return <AcknowledgeFavoriteAdd />
+
+	return <AddFavoriteForm onSubmit={saveTransfer} addFavoriteStore={addFavoriteStore} />
+}
+
+type SaveFormProps = {
+	onSubmit: () => void
+	addFavoriteStore: Signal<Partial<FavoriteModel> | undefined>
+}
+
+const AddFavoriteForm = ({ addFavoriteStore, onSubmit }: SaveFormProps) => {
+	if (addFavoriteStore.value === undefined) return <></>
+
+	const submitForm = (e: Event) => {
+		e.preventDefault()
+		onSubmit()
+	}
+
+	return (
+		<div class='my-4'>
+			<div class='font-bold text-2xl mb-2'>Save Transfer</div>
+			<div class='border border-dashed border-white/30 p-4'>
+				<div class='flex flex-col md:flex-row-reverse items-center gap-4'>
+					<div class='shrink w-full'>
+						<p class='text-white/50 text-sm'>Prevent accidental inputs by saving this transfer so you can quickly do this again later. Add a label to this transfer and hit save to continue.</p>
```

Comment:
I feel the label should be optional

Reply:
Let me track as discussion for now in #145